Assets/Scripts/UnitController.cs

```diff
@@ -29,6 +45,17 @@ public void Setup(int totalPosIndex, Vector2Int squadIndex)
     void FixedUpdate()
     {
         UnitMovement();
+
+        if (BattleManager.Instance.currentSquadSelection >= 0)
```

Comment:
Check the PlayerController script's `PlayerControls` method, you'll find the input handling for changing squads. You can essentially remove the usage of the `selectedSquad` variable and just use BattleManager's `currentSquadSelection` instead

Reply:
I will look at it

Reply:
Done, what do you think about it ?

Reply:
Works like a charm, good job!